packages/app-store/sendgridothercalendar/api/_postAdd.ts

```diff
@@ -0,0 +1,39 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { symmetricEncrypt } from "@calcom/lib/crypto";
+import { HttpError } from "@calcom/lib/http-error";
+import logger from "@calcom/lib/logger";
+import { defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+import checkSession from "../../_utils/auth";
+import getInstalledAppPath from "../../_utils/getInstalledAppPath";
+
+export async function getHandler(req: NextApiRequest, res: NextApiResponse) {
+  const session = checkSession(req);
+
+  const { api_key } = req.body;
+  if (!api_key) throw new HttpError({ statusCode: 400, message: "No Api Key provoided to check" });
+
+  const encrypted = symmetricEncrypt(JSON.stringify({ api_key }), process.env.CALENDSO_ENCRYPTION_KEY || "");
+
+  const data = {
+    type: "sendgrid_other_calendar",
+    key: { encrypted },
+    userId: session.user?.id,
```

Comment:
Just thinking out loud here, since I see in other places we do this as well. But isn't it actually weird that the `userId` is optional here? Because what would happen if you added null/undefined into that field? That wouldn't result in a successful addition of the credentials, right? Even though Prisma accepts it and won't error.

Reply:
Looking at this too - I agree it is weird and its strange that typescript doesn't recognise this as being not nullable	as in the `checkSession` function we do this https://github.com/calcom/cal.com/blob/6238b6d6b9d4f39d702c67cdf7788423ab4d546a/packages/app-store/_utils/auth.ts#L5

Reply:
Yeah, we might need to review this behavior to make sure there is no edge case that might force to an undesired app state. Resolving for this PR but @JeroenReumkens @sean-brydon feel free to open an issue to track it down.

Reply:
Looking at it - I think its an impossible edge case to ever hit. Should be okay on this front